build-utils/language-c/src/main/java/org/apache/plc4x/language/c/CLanguageTemplateHelper.java

```diff
@@ -288,7 +299,7 @@ public String getTypeSizeForField(TypedField field) {
 
     public String escapeValue(TypeReference typeReference, String valueString) {
         if (valueString == null) {
-            return null;
+            return "NULL";
```

Comment:
Shouldn't that than be a empty string ""?
```suggestion
            return "";
```
(Dislaimer: not a C expert ;)

Reply:
At this point it could be any string, this only gets used once in enum-template-c line 95 and it just checks for a "-1" value.
As long as it doesn't return an actual null we should be good.